bundles/pubsub/pubsub_spi/include/pubsub_utils_url.h

```diff
@@ -0,0 +1,55 @@
+/*
```

Comment:
Why this generic file? Isn't this admin specific? PS doesn't have to dictate how an admin uses urls.

Reply:
No this is not admin specific. This can be used in TCP / UDP and websocket admin.
ZMQ does do this already internally for the big part.

Reply:
Ah ok, I get it, since it is used by the other Admins as well I guess this makes sense. My first impression was that since it is in the spi, it would be a requirement for other admins as well.
Thinking out loud now, does it make sense to move these kind of functions to another directory? Technically it is not a SPI.

Reply:
@pnoltes @rbulter what do you guys think? Does it make sense to move this shared code to something like pubsub_shared? I agree it is generic enough to have in here, but I don't thinks it should be part of spi.

Reply:
I propose to name it pubsub_utils, the other pubsub_utils functions can also move there.

Reply:
Sounds good.

Reply:
I agree move it to pubsub_utils